src/components/CustomizedPipeline/__tests__/CustomizeAllPipelines.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import '@testing-library/jest-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { render } from '@testing-library/react';
+import CustomizeAllPipelines from '../CustomizeAllPipelines';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  useK8sWatchResource: jest.fn(() => [[], true]),
+}));
+
+jest.mock('react-router-dom', () => ({
+  Link: (props) => <a href={props.to}>{props.children}</a>,
+}));
+
+const useK8sWatchResourceMock = useK8sWatchResource as jest.Mock;
+
+describe('CustomizeAllPipelines', () => {
+  it('should render nothing while loading', () => {
+    useK8sWatchResourceMock.mockReturnValueOnce([[], false]);
+    const result = render(<CustomizeAllPipelines applicationName="" namespace="" />);
+    expect(result.baseElement.textContent).toBe('');
+  });
+
+  it('should render empty state', () => {
+    useK8sWatchResourceMock.mockReturnValueOnce([[], true]);
+    const result = render(<CustomizeAllPipelines applicationName="" namespace="" />);
+    expect(result.getByText('No components')).toBeInTheDocument();
+  });
+
+  it('should render empty state', () => {
+    useK8sWatchResourceMock.mockReturnValueOnce([
+      [
+        {
+          metadata: {
+            name: 'my-component',
+          },
+          spec: {
+            application: 'test',
+            source: {
+              git: {
+                url: 'https://github.com/org/test',
+              },
+            },
+          },
+        },
+      ],
+      true,
+    ]);
+    const result = render(<CustomizeAllPipelines applicationName="test" namespace="" />);
+    expect(result.getByTestId('component-row', { exact: false })).toBeInTheDocument();
+  });
+
+  it('should render empty state', () => {
```

Comment:
test names needs to be updated in this file.

Reply:
whoops, missed those. Updated